### ISSUE TITLE
Add referring entities from a Referenced By section (reverse-association)

### ISSUE THREAD

**Issue by rreganjr** [rreganjr/Requel#24]
## Add referring entities from a "Referenced By" section (reverse-association)

_Rewrites the original 2022 request ("Edit Goal: ability to assign Referring Entities from here") to reflect the app as it stands and the agreed generalized scope._

### Background

Original report: *"When adding a goal you can't associate it to an actor or stakeholder. It's tedious to have to go to the actor to add the goal there."*

Still true today. Associations are **owned by the referring container**: an actor/stakeholder/story/use-case each carry an explicit child list (`goals`, `actors`, `stories`), and the referenced entity only carries the inverse `referencedBy` for display. The goal editor now shows a **read-only "Referenced By"** table, but the only way to *create* the association is still from the container (e.g. the actor editor's "Goals" section). So the reverse-add the ticket asked for is unbuilt — just cheaper now, because the pieces exist (`entity-selector-dialog`, `app-relationship-section`, and the container-side "add child" commands).

### Scope (agreed: generalize, all referring types)

Make **"add a referring entity from a Referenced By section"** a shared capability, not goal-specific. Wherever an editor shows a Referenced By whose relationship is an **explicit, container-owned association**, offer an **Add** that:

1. opens the existing `entity-selector-dialog` to pick a referring container of an eligible type, then
2. creates the association by calling that container's existing **add-child** command with the current entity (no new backend command — reuse `container.add<Child>`).

Reverse-add pairs to support (target entity's Referenced By → addable referrer → command reused):

- **Goal** → Actor / Stakeholder / Story / Use-Case  (reverse of `container.goals`)
- **Actor** → Story / Use-Case  (reverse of `story.actors` / `useCase.actors`)
- (extend to any other editor that grows an explicit-association Referenced By)

**Out of scope:** the **glossary term** "Referenced By" — a term's references are derived from where the term appears in text, not a manual association, so it stays read-only. Any Referenced By that isn't a stored container-owned list is display-only.

### Approach

- A small, shared "add referring entity" affordance on the Referenced By section (mirrors the existing `app-relationship-section` add pattern), gated by `canEdit` and by whether the target is persisted.
- A mapping of `(targetType, referrerType) → container add-child command`, so the goal editor picks an actor and calls `actor.addGoal(thisGoal)`, etc. Enumerate the mapping from the models' child lists.
- Reuse `entity-selector-dialog` for picking (with type filtering when more than one referrer type is eligible), matching the actor editor's existing goal-linking UX.
- Refresh the Referenced By list after add; support remove symmetrically where it's cheap, or keep remove on the container side for now (decide during planning).

### Acceptance criteria

- From the **goal** editor you can associate an Actor, Stakeholder, Story, or Use-Case without leaving the goal; it appears in Referenced By and on the container's own list.
- From the **actor** editor you can add a referring Story / Use-Case the same way.
- The picker excludes already-associated and self entries; permissions (`canEdit`) and unsaved-entity states are respected.
- Glossary term Referenced By stays read-only (derived references).
- Existing container-side association flows and specs are unchanged; new behavior is covered by unit tests.

### Notes / open questions for planning

- Confirm each container exposes an add-child command reachable from the referenced side's service (actors add goals today; verify stakeholder/story/use-case equivalents and stakeholder's Referenced-By surfacing).
- Decide whether remove is offered from the Referenced By side too, or stays container-owned.
- Likely **frontend-mostly**; flag any missing backend command during planning.

Project: Requel 2.0. Suggest milestone **v2.0** and labels **ui-ux-review** + an **enhancement**/**feature** label (grown well past the original one-liner).

**Comment by rreganjr** [comment]
Shipped in #237, merged to release/2.0. Reverse-association ('Referenced By' add/remove) across the Goal, Actor, Story and Scenario editors, plus the goal-editor relation-section cleanup and referencedBy backend/test coverage.
